script/version_up.py

```diff
@@ -0,0 +1,660 @@
+  upgrade, by informing whether or not the target version is higher.
+
+  Args:
+    target_version:
+      The version being upgraded to.
+    working_version:
+      The current version of the current branch.
+  """
+    to_version = [int(i) for i in target_version.split('.')]
+    in_version = [int(i) for i in working_version.split('.')]
+
+    if len(in_version) != 4 or len(to_version) != 4:
+        return False
+
+    for origin, target in zip(in_version, to_version):
+        if target > origin:
+            return True
+        if target < origin:
+            return False
+
+    return False
+
+
+def has_staged_changed():
+    return _run_git('diff', '--cached', '--stat') != ''
+
+
+class Upgrade:
+    """The upgrade process, holding the data related to the upgrade.
+
+  This class produces an object that is reponsible for keeping track of the
+  upgrade process step-by-step. It acquires all the common data naecessary for
+  its completion.
+  """
+
+    def __init__(self, base_branch, target_version):
+        self.base_branch = base_branch
+        self.target_version = target_version
+        self.base_version = _get_chromium_version_from_git(base_branch)
+        self.working_version = _get_chromium_version_from_git('HEAD')
+
+    def _update_package_version(self):
+        """Creates the change upgrading the Chromium version
+
+    This is for the creation of the first commit, which means updating
+    package.json to the target version provided, and commiting the change to
+    the repo
+    """
+        package = _load_package_file('HEAD')
+        package['config']['projects']['chrome']['tag'] = self.target_version
+        with open(PACKAGE_FILE, "w") as package_file:
+            json.dump(package, package_file, indent=2)
+
+        _run_git('add', PACKAGE_FILE)
+
+        # Pinlist timestamp update occurs with the package version update.
+        _update_pinslist_timestamp()
+        _run_git('add', PINSLIST_TIMESTAMP_FILE)
+
+        _run_git(
+            'commit', '-m',
+            f'Update from Chromium {self.base_version} to Chromium {self.target_version}.'
+        )
+
+        commit = _run_git('log', '-1', '--pretty=oneline', '--abbrev-commit')
+        print(f'Done: {commit}')
+
+    def _save_updated_patches(self):
+        """Creates the updated patches change
+
+    This function creates the third commit in the order of the update, saving
+    all patches that might have been changed or deleted. Untracked patches are
+    excluded from addition at this stage.
+    """
+        _run_git('add', '-u', '*.patch')
+
+        if has_staged_changed() is False:
+            # Nothing to commit
+            return
+        _run_git(
+            'commit', '-m',
+            f'Update patches from Chromium {self.base_version} to Chromium {self.target_version}.'
+        )
+
+        commit = _run_git('log', '-1', '--pretty=oneline', '--abbrev-commit')
+        print(f'Done: {commit}')
```

Comment:
For future improvement: this could be a function that takes the commit message and returns commit hash.

Reply:
I'll add this the next run I have so I can test it.

Reply:
Done.